app/Services/GraphQL.php

```diff
@@ -15,14 +16,49 @@ public function __construct()
     }
 
     /**
-     * Run a GraphQL query using the client.
+     * Run a GraphQL query using the client and return the data result.
      *
      * @param  $query     String
      * @param  $variables Array
-     * @return \Softonic\GraphQL\Response
+     * @return array|null
      */
     public function query($query, $variables)
     {
-        return $this->client->query($query, $variables);
+        // Use try/catch to avoid any GraphQL related errors breaking the application.
+        try {
+            $response = $this->client->query($query, $variables);
+        } catch (\Exception $exception) {
+            Log::error(
+                'GraphQL request failed. Query: '.$query
```

Comment:
Oh, I didn't think of this before but let's put these variables into the "meta" block rather than stringifying:

```php
Log::error('GraphQL request failed.', ['query' => $query, 'variables' => $variables, 'exception' => $exception->getMessage()]);
```

Reply:
Oh sweet, way cleaner!! Updated in https://github.com/DoSomething/rogue/pull/889/commits/53c090d550046cf0b04d9a7becdb427405da48b8